content/blueprints/gcp.md

```diff
@@ -0,0 +1,1277 @@
+    unless overridden by a defined zone/subnetwork.
+  * `auth` the JSON key file provided by GCP.
+    Can either be the contents of the JSON file, or a file path.
+    This should be in the format provided by the GCP credentials JSON export (https://developers.google.com/identity/protocols/OAuth2ServiceAccount#creatinganaccount)
+  * (optional) `network` the default network to place network-scoped nodes in.
+    The default network (`default`) will be used if this is not specified.
+
+Example
+{{< gsHighlight  yaml  >}}
+...
+node_types:
+  my_vm:
+    type: cloudify.gcp.nodes.Instance
+    properties:
+      image_id: <GCP image ID>
+      gcp_config:
+        project: a-gcp-project-123456
+        zone: us-east1-b
+        auth: <GCP auth JSON file>
+{{< /gsHighlight >}}
+
+
+## Using Existing Resources
+
+Many Cloudify GCP types have a property named `use_external_resource`, which defaults to `false`. When set to `true`, the plugin will apply different semantics for each of the operations executed on the relevant node's instances:
+
+  If `use_external_resource` is `true`, then the required properties for that type (`name`, possibly `region` or `zone`) will be used to look up an existing entity in the GCP project.
+  If the entity is found, then its data will be used to popluate the cloudify instance's attributes (`runtime_properties`). If it is not found then the blueprint will fail to deploy.
+
+
+This behavior is common to all resource types which support `use_external_resource`:
+
+ * `create` If `use_external_resource` is true, the GCP plugin will check if the resource is available in your account. If no such resource is available, the operation will fail, if it is available, it will assign the resource details to the instance `runtime_properties`.
+ * `delete` If `use_external_resource` is true, the GCP plugin will check if the resource is available in your account. If no such resource is available, the operation will fail, if it is available, it will unassign the instance `runtime_properties`.
+
+
+## Runtime Properties
+
+See section on [runtime properties](http://cloudify-plugins-common.readthedocs.org/en/3.3/context.html?highlight=runtime#cloudify.context.NodeInstanceContext.runtime_properties)
+
+Most node types will write a snapshot of the `resource` information from GCP when the node creation has finished (some, e.g. DNSRecord don't correspond directly to an entity in GCP, so this is not universal).
+
+
+# Node Types
+
+## cloudify.gcp.nodes.Address
+**Derived From:** cloudify.gcp.nodes.GlobalAddress
+
+A GCP Address. This can be connected to an Instance using the `cloudify.gcp.relationships.instance_connected_to_ip` relationship type
+
+
+
+**Properties:**
+
+
+  * `gcp_config`
+    A dictionary of values to pass to authenticate with the GCP API.
+
+    *default:* {}
+  * `region`
+    Region to place the Address in. If not provided it defaults to the value in `gcp_config` (which defaults to 'default').
```

Comment:
Region **in which** to place the Address ~~ in ~~. If not provided **,** it defaults